nylon-threads/src/main/java/com/palantir/nylon/threads/VirtualThreads.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.nylon.threads;
+
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Utility functionality to allow libraries which compile for earlier JDKs
+ * to take advantage of virtual threads features on sufficiently new runtimes.
+ */
+public final class VirtualThreads {
+    private static final SafeLogger log = SafeLoggerFactory.get(VirtualThreads.class);
+
+    @Nullable
+    private static final VirtualThreadSupport VIRTUAL_THREAD_SUPPORT = maybeInitialize();
+
+    public static boolean isVirtual(Thread thread) {
+        return VIRTUAL_THREAD_SUPPORT != null && VIRTUAL_THREAD_SUPPORT.isVirtual(thread);
+    }
```

Comment:
I'd like to use this utility function in tritium to tag thread metrics with `thread: platform` or `thread: virtual`